test/e2e_node/pod_host_ips.go

```diff
@@ -158,41 +123,34 @@ var _ = common.SIGDescribe("Dual Stack Host IP [Feature:PodHostIPs]", func() {
 			framework.ExpectNoError(err)
 			for _, node := range nodeList.Items {
 				if node.Name == p.Spec.NodeName {
-					nodeIPs := []string{}
+					got := sets.New[string]()
+					for _, hostIP := range p.Status.HostIPs {
+						got.Insert(hostIP.IP)
+					}
+
+					want := sets.New[string]()
 					for _, address := range node.Status.Addresses {
 						if address.Type == v1.NodeInternalIP {
-							nodeIPs = append(nodeIPs, address.Address)
+							want.Insert(address.Address)
 						}
 					}
-					gomega.Expect(p.Status.HostIPs).Should(gomega.Equal(nodeIPs))
```

Comment:
I think this is the problem expressed here https://github.com/kubernetes/kubernetes/commit/6d02fbdbe07422b1d48fc20dc4ba7d3f98603a95#r130786064

Reply:
Yeah, I don't remember why I wrote it that way, because in the alpha phase, the initial merge was a revert and then a merge over a bit of a long period of time, and I didn't realize that until I switched to the beta phase and ran the test this time.